editor/src/components/canvas/canvas-strategies/strategies/reparent-helpers/reparent-strategy-parent-lookup.ts

```diff
@@ -0,0 +1,482 @@
+  )
+
+  const containingComponentsUnderMouse = containingComponents.filter((c) =>
+    allElementsUnderPoint.find((p) => EP.pathsEqual(c, p)),
+  )
+
+  return containingComponentsUnderMouse.some((c) => EP.isDescendantOf(c, target))
+}
+
+function findParentByPaddedInsertionZone(
+  metadata: ElementInstanceMetadataMap,
+  validTargetparentsUnderPoint: ElementPath[],
+  canvasScale: number,
+  pointOnCanvas: CanvasPoint,
+) {
+  let targetParentWithPaddedInsertionZone: ReparentTarget | null = null
+  const singleAxisAutoLayoutContainersUnderPoint = mapDropNulls((element) => {
+    const autolayoutDirection = singleAxisAutoLayoutContainerDirections(element, metadata)
+    if (autolayoutDirection === 'non-single-axis-autolayout') {
+      return null
+    }
+    const shouldReparentAsFlowOrStatic = autoLayoutParentAbsoluteOrStatic(metadata, element)
+    if (shouldReparentAsFlowOrStatic === 'REPARENT_AS_ABSOLUTE') {
+      return null
+    }
+    const compatibleWith1DReorder = isSingleAxisAutoLayoutComaptibleWithReorder(metadata, element)
+    if (!compatibleWith1DReorder) {
+      return null
+    }
+
+    return {
+      path: element,
+      directions: autolayoutDirection,
+    }
+  }, [...validTargetparentsUnderPoint].reverse())
+
+  // first try to find a flex element insertion area
+  for (const singleAxisAutoLayoutContainer of singleAxisAutoLayoutContainersUnderPoint) {
+    const { direction, forwardsOrBackwards, flexOrFlow } = singleAxisAutoLayoutContainer.directions
+
+    const targets: Array<{ rect: CanvasRectangle; insertionIndex: number }> =
+      drawTargetRectanglesForChildrenOfElement(
+        metadata,
+        singleAxisAutoLayoutContainer.path,
+        'padded-edge',
+        canvasScale,
+        direction,
+        forwardsOrBackwards,
+      )
+
+    const foundTarget = targets.find((target) => {
+      return rectContainsPoint(target.rect, pointOnCanvas)
+    })
+    const targetUnderMouseIndex = foundTarget?.insertionIndex
+
+    if (targetUnderMouseIndex != null) {
+      // we found a target!
+      targetParentWithPaddedInsertionZone = {
+        shouldReparent: true,
+        shouldReorder: true,
+        newParent: singleAxisAutoLayoutContainer.path,
+        newIndex: targetUnderMouseIndex,
+        shouldConvertToInline:
+          flexOrFlow === 'flex' || direction == null ? 'do-not-convert' : direction,
+        defaultReparentType: 'REPARENT_AS_STATIC',
+      }
+    }
+  }
+  return targetParentWithPaddedInsertionZone
+}
+
+function findParentUnderPointByArea(
+  targetParentPath: ElementPath,
+  metadata: ElementInstanceMetadataMap,
+  canvasScale: number,
+  pointOnCanvas: CanvasPoint,
+) {
+  const autolayoutDirection = singleAxisAutoLayoutContainerDirections(targetParentPath, metadata)
+  const shouldReparentAsFlowOrStatic = autoLayoutParentAbsoluteOrStatic(metadata, targetParentPath)
+  const compatibleWith1DReorder = isSingleAxisAutoLayoutComaptibleWithReorder(
```

Comment:
not every single axis autolayout is compatible with 1d reorder. specifically, for Flow if there's only a single children, we can't use that to determine direction, and in our testing with @gbalint we realized that the behavior just becomes unintuitive overall. so we only allow 1d reorder-like insertion if there's >1 static children for flow parents. for flow parents that have 0 or 1 children, we still allow static reorder, but we don't run the insertion indicators and the padded area checks

Reply:
this is the same behavior as on master today, but on master this is totally implicit in https://github.com/concrete-utopia/utopia/blob/0452ce132ca5a1524e8685403d6922297dcdff8d/editor/src/components/canvas/canvas-strategies/strategies/reparent-strategy-helpers.ts#L1062-L1067